components/fetch-build-scan-data-cmdline-tool/src/main/java/com/gradle/enterprise/api/client/GradleEnterpriseApiClient.java

```diff
@@ -110,6 +137,18 @@ public BuildValidationData fetchBuildValidationData(String buildScanId) {
                     throw new BuildScanNotFoundException(buildScanId, baseUrl, e.getCode(), e.getResponseBody(), e);
                 case StatusCodes.UNAUTHORIZED:
                     throw new AuthenticationFailedException(buildScanId, baseUrl, e.getCode(), e.getResponseBody(), e);
+                case 0:
+                    // This wasn't a bad HTTP response, instead it was a failure to connect.
+                    //
+                    // WORKAROUND: There is a bug in the OpenAPI generator that generates an ApiException.getMessage() method
+                    // that throws a NullPointerException  when called on a non-HTTP error. See https://github.com/OpenAPITools/openapi-generator/issues/12631
+                    //
+                    // To work around the bug we use the underlying cause (if there is one). If there isn't a cause exception, then use
+                    // the ApiException but avoid calling its getMessage method.
+                    if (e.getCause() != null) {
+                        throw new ConnectionFailedException(buildScanId, baseUrl, e.getCause());
+                    }
+                    throw new ConnectionFailedException(buildScanId, baseUrl, e);
```

Comment:
I would use an `else` here to make it more explicit.

Reply:
✅ Done